### PR TITLE
refactor: extracted starter project code from create options

### PR DESCRIPTION
**Pull request by girishramnani**
**What type of PR is this?**
 /kind code-refactoring
**What does does this PR do / why we need it**:

- a subpackage shouldn't use its parent package like `component/ui` shouldn't use `component` package
- refactoring starter project code in to `component` package so it can be unit tested

**Which issue(s) this PR fixes**:

Fixes #?

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [x] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: pkg/component/starter_project.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+		}
+	}
+
+	return project, err
+
+}
+
+// Downloads first starter project from list of starter projects in devfile
+// Currently type git with a non github url is not supported
+func DownloadStarterProject(devObj parser.DevfileObj, starterProject *devfilev1.StarterProject, decryptedToken string, registryName string) error {
```
Comment (girishramnani) [inline pkg/component/starter_project.go]: I am sensing that this function could be part of `devfilev1.StarterProject` struct as a method, but it then needs to land in the api repo

File: pkg/odo/cli/component/create_helpers.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	envs, err := config.NewEnvVarListFromSlice(co.componentEnvVars)
+	if err != nil {
+		return err
+	}
+	co.componentSettings.Envs = envs
+	co.ignores = []string{}
+	return nil
+}
+
+func DecideAndDownloadStarterProject(devObj parser.DevfileObj, projectPassed string, token string, interactive bool) error {
```
Comment (mik-dass) [inline pkg/odo/cli/component/create_helpers.go:41]: Please add some comments here

File: pkg/component/starter_project.go [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+			return nil, errors.Errorf("the project: %s specified in --starter does not exist", projectPassed)
+		}
+	}
+
+	return project, err
+
+}
+
+// Downloads first starter project from list of starter projects in devfile
+// Currently type git with a non github url is not supported
```
Comment (mik-dass) [inline pkg/component/starter_project.go:75]: We have moved to go-git and currently support non-github urls.

File: pkg/odo/cli/component/ui/ui.go [hunk 4]
```diff
@@ -218,6 +219,22 @@ func createComponentNameValidator(context *genericclioptions.Context) survey.Val
...
+// Exists checks whether a component with the given name exists in the current application or not
+// componentName is the component name to perform check for
+// The first returned parameter is a bool indicating if a component with the given name already exists or not
+// The second returned parameter is the error that might occurs while execution
+func exists(client *occlient.Client, componentName, applicationName string) (bool, error) {
+	deploymentName, err := util.NamespaceOpenShiftObject(componentName, applicationName)
+	if err != nil {
+		return false, errors.Wrapf(err, "unable to create namespaced name")
+	}
+	deployment, _ := client.GetDeploymentConfigFromName(deploymentName)
```
Comment (mik-dass) [inline pkg/odo/cli/component/ui/ui.go:231]: Why are we using occlient directly instead of `component.Exists()`?

Comment (girishramnani) [reply pkg/odo/cli/component/ui/ui.go]: addressed

File: pkg/component/starter_project.go (hunk 3 above)
Comment (girishramnani) [reply pkg/component/starter_project.go]: removed the comment

File: pkg/odo/cli/component/create_helpers.go (hunk 2 above)
Comment (girishramnani) [reply pkg/odo/cli/component/create_helpers.go:43]: added

**Comment by mik-dass** [comment]
```
        I0122 07:18:50.432981   50107 preference.go:222] The path for preference file is /tmp/010926755/preference.yaml
         ✗  Folder /tmp/010926755 contains one element and it's not the devfile used.
        
    to contain substring
        <string>: no starter project found in devfile.
    /go/src/github.com/openshift/odo/tests/helper/helper_generic.go:70
```
Some error occurred on the CI, please have a look.

**Comment by mik-dass** [comment]
> Some error occurred on the CI, please have a look.

Fix in https://github.com/openshift/odo/pull/4384/

File: pkg/odo/cli/component/create_helpers.go (hunk 2 above)
Comment (mik-dass) [reply pkg/odo/cli/component/create_helpers.go:43]: Can we make this a private function?

Comment (girishramnani) [reply pkg/odo/cli/component/create_helpers.go:43]: done

**Review by mik-dass — APPROVED** [review]
/lgtm

**Pull request merged**